yunwoo/programmers-1/약수의-개수와-덧셈.js

```diff
@@ -0,0 +1,49 @@
+function solution(left, right) {
+  let result = 0;
+  const numArr = new Array(right - left + 1).fill().map((_, i) => left + i);
+
+  for (let element of numArr) {
+    let i = 1;
+    const divisorArr = [];
+
+    while (i <= element) {
+      if (element % i === 0) {
+        divisorArr.push(i);
+      }
+      i++;
+    }
+
+    !(divisorArr.length % 2) ? (result += element) : (result -= element);
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(left, right) {
  const numArr = Array.from({ length: right - left + 1}, (_, i) => left + i);
  let result = 0;

  for (let element of numArr) {
    let divisorCount = 0;
    let i = 1;

    while (i <= element) {
      if (element % i === 0) divisorCount++;
      i++;
    }

    result += !(divisorCount % 2) ? element : -element;
  }

  return result;
}
```
- 풀이 잘봤습니다 🙇‍♂️ 항상 상수를 의미하는 `const가 let보다 상단에 위치`할 수 있게 해주세요 :) 일반적으로 코딩을 할 때 상수를 파일 상단으로 올려주는 암묵적인 룰이 있습니다
- new Array하고 fill().map()보다 그냥 `Array.from`을 활용해서 배열을 생성해주는게 좋아보입니다. `불 필요한 메서드 체이닝을 제거`해줍니다
- 또한, 해당 풀이에서 while문 내에서 `i`를 활용하는 상황이니 그냥 `for문`을 활용하는게 더 좋을 것 같습니다 ~!
- divisorArr에다 요소를 push하고 이 배열의 길이를 판단하기보다, 단순히 카운트로 관리하는 방법도 참고해주세요 🙆‍♂️
- result에 누적할 때도 `조건 ? (result += element) : (result -= element)` 보다 `result += 조건 ? element : -element`가 좀 더 깔끔하다는 생각이 들었습니다 🙆‍♂️

Reply:
- 코딩컨벤션 더 신경쓰겠습니다🥲
- `Array.from`에 lenght 와 콜백함수를 이용한 방법이 있었군요!! 감사합니다ㅎㅎ
- 저도 작성하고 보니 for문이 더 좋을거 같다는 생각이 들었습니다!!🙆‍♂️
- 대입연산자를 한번만 썼어도 되는군요..! 다음엔 꼭 활용해보겠습니다 리뷰 감사합니다🙇‍♂️!!!!